crates/ruff/src/autofix/helpers.rs

```diff
@@ -444,6 +447,66 @@ pub fn remove_argument(
     }
 }
 
+/// Generate an [`Edit`] to reference the given symbol. Returns the [`Edit`] necessary to make the
+/// symbol available in the current scope along with the bound name of the symbol.
+///
+/// For example, assuming `module` is `"functools"` and `member` is `"lru_cache"`, this function
+/// could return an [`Edit`] to add `import functools` to the top of the file, alongside with the
+/// name on which the `lru_cache` symbol would be made available (`"functools.lru_cache"`).
+///
+/// Attempts to reuse existing imports when possible.
+pub fn get_or_import_symbol(
+    module: &str,
+    member: &str,
+    context: &Context,
+    importer: &Importer,
+    locator: &Locator,
+) -> Result<(Edit, String)> {
+    if let Some((source, binding)) = context.resolve_binding(module, member) {
+        // If the symbol is already available in the current scope, use it, and add a no-nop edit to
+        // force conflicts with any other fixes that might try to remove the import.
+        let import_edit = Edit::replacement(
+            locator.slice(source).to_string(),
+            source.location,
+            source.end_location.unwrap(),
+        );
```

Comment:
This piece here is a hack. Consider:

```py
import sys

quit()
```

Assume you _omit_ this hack. If you run with `unused-imports` and `sys-exit-alias` here, then we'll generate two fixes: (1) we'll remove the unused `import sys`; (2) we'll replace `quit()` with `sys.exit()`, under the assumption that `sys` is already imported and available. That'd leave us with an invalid reference.

The workaround is to add a "fake" edit to replace `import sys` with `import sys`. This ensures that we don't try to apply conflicting autofixes to the file.

We may be getting to the point at which we want to avoid trying to apply multiple fixes in the same autofix pass (which would also avoid this issue).